pkg/reconciler/route/resources/ingress.go

```diff
@@ -76,7 +77,38 @@ func MakeClusterIngress(
 	}, nil
 }
 
-func makeIngressSpec(
+// MakeIngress creates Ingress to set up routing rules. Such Ingress specifies
+// which Hosts that it applies to, as well as the routing rules.
+func MakeIngress(
+	ctx context.Context,
+	r *servingv1alpha1.Route,
+	tc *traffic.Config,
+	tls []v1alpha1.IngressTLS,
+	clusterLocalServices sets.String,
+	ingressClass string,
+) (v1alpha1.IngressAccessor, error) {
+	spec, err := MakeIngressSpec(ctx, r, tls, clusterLocalServices, tc.Targets)
+	if err != nil {
+		return nil, err
+	}
+	return &v1alpha1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      names.Ingress(r),
+			Namespace: r.Namespace,
+			Labels: map[string]string{
+				serving.RouteLabelKey:          r.Name,
+				serving.RouteNamespaceLabelKey: r.Namespace,
+			},
+			Annotations: resources.UnionMaps(map[string]string{
+				networking.IngressClassAnnotationKey: ingressClass,
+			}, r.ObjectMeta.Annotations),
+		},
```

Comment:
The Ingress should have an `OwnerReference` to the `Route`.  This is not possible for `ClusterIngress`, which is why we have the labels.

Basically you need a line like this:
https://github.com/knative/serving/blob/936b4b05e1dca3b1a4db6a2a60933e0abde4bcb0/pkg/reconciler/revision/resources/deploy.go#L233

Reply:
Fixed.